### DISCUSSION TITLE
Installing and running WSA offline (without Internet)

### DISCUSSION THREAD

**Discussion by Bob3500** [microsoft/WSA#459] [General]
Dear all,

I'm facing a particular problem and maybe someone can help me on this :

For a specific project, I need to install and run WSA on a Windows 11 computer without Internet access (resources are accessible only on a private local network).

To do this, I use the "unoffical" way to download and install the WSA msixbundle, and then sideload the APK of my Android app.
Everything is ok, when doing this while having an access to Internet.

Without Internet access, the result is the following : 
- WSA msixbundle installation : ok, no problem.
- but, when I launch the 'Windows Subsystem for Android™ Settings' app, it opens and closes immediately. As if a dependency was missing or a check (request ?) should require an access to Internet at runtime.
- If I give back Internet to the computer, then it's going better : the 'Windows Subsystem for Android™ Settings' app is successfully launched. After that I can cut back Internet, it continues to work.

The problem is that I absolutely have to work without Internet (from start to finish) !! Any ideas ?

Thank you for your support.

**Comment by marshallrichards** [comment]
Hi @Bob3500 , did you ever solve this?

**Reply by Bob3500** [reply]
Hi @marshallrichards,
Partially : I have tested the WSA packaging "WSABuilds" provided by MustardChef (https://github.com/MustardChef/WSABuilds), and it's working fine without Internet. But I didn't spend time to understand what was blocking.

I have interrupted this work when Microsoft decided to kill WSA.

I suspect this behavior could be used in 2025 by Microsoft to shut down WSA for good.

**Reply by marshallrichards** [reply]
@Bob3500 Thanks for getting back to me so quickly!
Just to clarify, were you able to install the WSA packaging from [WSABuilds](https://github.com/MustardChef/WSABuilds) without internet, like completely offline after installing Windows 11?

**Reply by Bob3500** [reply]
Yes, that's it. Completely offline.
